main.go

```diff
@@ -1014,6 +1014,60 @@ func processImportCfg(flags []string) (newImportCfg string, _ error) {
 	return newCfg.Name(), nil
 }
 
+type knownReflect struct {
+	pkgPath string
+	name    string
+}
+
+type funcFullName = string
+
+// knownReflectAPIs is a static record of what std APIs use reflection on their
+// parameters, so we can avoid obfuscating types used with them.
+//
+// For now, this table is manually maintained, until we automatically detect and
+// propagate the uses of reflect in std and third party APIs.
+// If we end up maintaining this table for a long time, we should probably
+// improve the process via either code generation or sanity checks.
+//
+// TODO: record which parameters get used with reflection, as it's often not all
+// of them.
+//
+// TODO: we're not including fmt.Printf, as it would have many false positives,
+// unless we were smart enough to detect which arguments get used as %#v or %T.
+//
+// TODO: this should also support third party APIs; see
+// https://github.com/burrowers/garble/issues/162
+var knownReflectAPIs = map[funcFullName]bool{
```

Comment:
I can think of a few more functions/methods in the stdlib that use reflection, ie encoding/binary, encoding/asn1 as of Go 1.16 at least, haven't checked if there are more. I agree we should have some kind of script or something to generate this list at some point, but for now this is good enough I think

Reply:
I imagined I missed a bunch :) I saw asn1 but thought most people aren't using it directly. I completely forgot about encoding/binary.

Perhaps do a follow-up PR after we merge this? I think it's going to be easier to collaborate right on master, than to coordinate how to join and co-author commits in single PRs.